zigpy/util.py

```diff
@@ -479,3 +485,109 @@ def __getattr__(name: str) -> typing.Any:
         return replacement
 
     return __getattr__
+
+
+@dataclasses.dataclass(frozen=True)
+class BaseRequestListener:
+    device: zigpy.device.Device
+    matchers: tuple[zigpy.typing.MatcherType]
```

Comment:
any reason this is a tuple and not a list? the other signatures are all accepting lists

Reply:
The listeners are supposed to be immutable, this should be converted into a tuple within `__post_init__`.